static/js/narrow.js

```diff
@@ -150,6 +151,23 @@ exports.activate = function (raw_operators, opts) {
     var filter = new Filter(raw_operators);
     var operators = filter.operators();
 
+    // Take the most detailed part of the narrow to use as the title.
+    // If the operator is something other than "stream", "topic", or
+    // "is", we shouldn't update the narrow title
+    if (filter.has_operator("stream")) {
+        if (filter.has_operator("topic")) {
+            exports.narrow_title = operators[1].operand;
+        } else {
+            exports.narrow_title = operators[0].operand;
+        }
+    } else if (filter.has_operator("is")) {
+        exports.narrow_title = operators[0].operand;
```

Comment:
I think that this should be: `filter.operands("topic")[0];` instead (and similarly for the other cases).  

Otherwise, this could result in weird behavior in the event that the user has an unusual sequence of operators.

Sorry I missed this previously!  Otherwise this change lgtm.